uxarray/grid/grid.py

```diff
@@ -2254,32 +2283,26 @@ def get_faces_at_constant_latitude(self, lat, use_spherical_bounding_box=False):
                 f"Latitude must be between -90 and 90 degrees. Received {lat}"
             )
 
-        if use_spherical_bounding_box:
-            faces = constant_lat_intersections_face_bounds(
-                lat=lat,
-                face_min_lat_rad=self.bounds.values[:, 0, 0],
-                face_max_lat_rad=self.bounds.values[:, 0, 1],
-            )
-            return faces
-        else:
-            edges = self.get_edges_at_constant_latitude(lat, use_spherical_bounding_box)
-            faces = np.unique(self.edge_face_connectivity[edges].data.ravel())
-
-            return faces[faces != INT_FILL_VALUE]
+        faces = constant_lat_intersections_face_bounds(
+            lat=lat,
+            face_bounds_lat=self.face_bounds_lat.values,
+        )
+        return faces
 
-    def get_edges_at_constant_longitude(self, lon, use_spherical_bounding_box=False):
+    def get_edges_at_constant_longitude(
+        self, lon: float, use_spherical_bounding_box: bool = False
+    ):
         """
         Identifies the indices of edges that intersect with a line of constant longitude.
 
         Parameters
         ----------
         lon : float
-            The longitude at which to identify intersecting edges, in degrees.
+            The longitude at which to extract the cross-section, in degrees.
+            Must be between -90.0 and 90.0
         use_spherical_bounding_box : bool, optional
-            If `True`,
-            computes the bounding box for each face using great circle arcs for edges
-            and considers extreme minimums or maximums to increase accuracy.
-            Defaults to `False`.
+            If True, uses a spherical bounding box for obtaining candidate faces, which considers the extreme cases
```

Comment:
Just for the information, we don't need to worry about the "curvature" of a great circle arc will affect the max/min longitude. This curvature thing will only affect the max/min latitude.

Reply:
Good catch. Forgot to update this section.